### PR TITLE
Enable event logging for cloud hypervisor

### PR DESCRIPTION
**Pull request by russell-islam**
Signed-off-by: Muminul Islam <muislam@microsoft.com>

Verified with creating VM and checked if the log file and log entry created accordingly.

### CODE REVIEW

File: src/ch/ch_domain.c [hunk 1]
```diff
@@ -40,6 +44,41 @@ VIR_ENUM_IMPL(virCHDomainJob,
...
+G_DEFINE_TYPE(chDomainLogContext, ch_domain_log_context, G_TYPE_OBJECT);
+//static void chDomainLogContextFinalize(GObject *obj);
+static void
+chDomainLogContextFinalize(GObject *object)
+{
+    chDomainLogContextPtr ctxt = CH_DOMAIN_LOG_CONTEXT(object);
+    VIR_DEBUG("ctxt=%p", ctxt);
+
+    virLogManagerFree(ctxt->manager);
+    VIR_FREE(ctxt->path);
```
Comment (bryteise) [inline src/ch/ch_domain.c]: This should use g_free, the VIR_FORCE_CLOSE are still in use upstream though.

File: src/ch/ch_domain.c [hunk 2]
```diff
@@ -40,6 +44,41 @@ VIR_ENUM_IMPL(virCHDomainJob,
...
+}
+static void ch_domain_log_context_class_init(chDomainLogContextClass *klass)
+{
+    GObjectClass *obj = G_OBJECT_CLASS(klass);
+
+    obj->finalize = chDomainLogContextFinalize;
+}
+
+G_DEFINE_TYPE(chDomainLogContext, ch_domain_log_context, G_TYPE_OBJECT);
+//static void chDomainLogContextFinalize(GObject *obj);
```
Comment (bryteise) [inline src/ch/ch_domain.c]: Should remove this I think.

**Comment by bryteise** [comment]
Just two little cleanups, otherwise looks ready to me.

File: src/ch/ch_domain.c (hunk 2 above)
Comment (russell-islam) [reply src/ch/ch_domain.c]: Fixed

File: src/ch/ch_domain.c (hunk 1 above)
Comment (russell-islam) [reply src/ch/ch_domain.c]: Fixed.

File: src/ch/ch_domain.c [hunk 3]
```diff
@@ -40,6 +44,41 @@ VIR_ENUM_IMPL(virCHDomainJob,
...
+
+    int writefd;
+    int readfd; /* Only used if manager == NULL */
+    off_t pos;
+    ino_t inode; /* Only used if manager != NULL */
+    char *path;
+    virLogManagerPtr manager;
+};
+static void chDomainLogContextFinalize(GObject *obj);
+static void ch_domain_log_context_init(chDomainLogContext *logctxt G_GNUC_UNUSED)
```
Comment (praveen-pk) [inline src/ch/ch_domain.c:58]: Is this empty method required?

File: src/ch/ch_monitor.c [hunk 4]
```diff
@@ -735,6 +736,10 @@ static int virCHMonitorProcessEvent(virCHMonitorPtr mon,
 
     ev = virCHMonitorEventFromString(source, event);
     VIR_DEBUG("Source: %s Event: %s, ev: %d", source, event, ev);
+    virCHDriverPtr driver = CH_DOMAIN_PRIVATE(vm)->driver;
+    /* log start of migration */
```
Comment (praveen-pk) [inline src/ch/ch_monitor.c]: Following link logs all Events from CLH, not just migration. Could you update the comment appropriately?

File: src/ch/ch_domain.c (hunk 3 above)
Comment (praveen-pk) [reply src/ch/ch_domain.c:58]: nvm confirmed Qemu also does this.

**Review by praveen-pk — APPROVED** [review]

**Pull request merged**